src/knot_cloud.c

```diff
@@ -639,12 +639,13 @@ int knot_cloud_list_devices(void)
  * Returns: 0 if successful and a KNoT error otherwise.
  */
 int knot_cloud_publish_data(const char *id, uint8_t sensor_id,
-			    uint8_t value_type, const knot_value_type *value,
+			    uint8_t value_type, void *value,
 			    uint8_t kval_len)
 {
 	json_object *jobj_data;
 	const char *json_str;
 	int result;
+	knot_value_type *aux = value;
```

Comment:
What do you think to do a explicit cast:
`knot_value_type *aux = (knot_value_type *) value`

ps.: this is just a suggestion